### ISSUE TITLE
MyLocationLayer: add an optional callout/label

### ISSUE THREAD

**Issue by janusw** [Mapsui/Mapsui#1143]
It would be useful to have a configurable callout-like functionality for `MyLocationLayer`, e.g. to show the coordinates of the user's position (or other information).

Technically, the problem seems to be that the `Callout` class requires a `Pin` to work with, and the location icon in `MyLocationLayer` is not a `Pin` (but a `Feature`). Possible solutions could include:
* Making the location icon in `MyLocationLayer` an actual `Pin`.
* Extend the `Callout` class to work not only with a `Pin`, but also with a `Feature`.
* Create a new class ('FeatureCallout'?) that works with a `Feature`.
* ...?

I have not checked if any of this is feasible at all (and if yes, which one is preferable).

**Comment by pauldendulk** [comment]
In the long run a solution that does not depend on Xamarin.Forms will be the better solution.

**Comment by janusw** [comment]
In principle one could simply use a GeometryFeature with a LabelStyle (cf. the LabelsSample).

The only problem I can see with this approach: It would be great to be able to toggle the label by clicking on the MyLocation icon, but a GeometryFeature as such is not really clickable, is it?

**Comment by pauldendulk** [comment]
> but a GeometryFeature as such is not really clickable, is it?

Yes, you can use the MapInfo for this. An example of how this is done is in the 'Info' | 'Single Callout' sample. Click on a dot. Perhaps this mechanism could be improved, but I think it is much better to develop in this direction. The dependencies on the platforms always cause problems in the long run, and this 'native' support would also work in other platforms such as Avalonia, Uno and WinUI. Also, more platform dependent code results in more maintenance, and more bugs (because we do not test extensively on every platform).

**Comment by janusw** [comment]
> > but a GeometryFeature as such is not really clickable, is it?
> 
> Yes, you can use the MapInfo for this. An example of how this is done is in the 'Info' | 'Single Callout' sample. Click on a dot. Perhaps this mechanism could be improved, but I think it is much better to develop in this direction.

Thanks for the comments, Paul, but I'm afraid you might slightly misunderstand the scope of what I'm proposing:
1. I don't want to display information outside of the map (this is what the sample does that you mention, and I can already achieve that). I want information to be displayed in a label on the map itself, next to the user's location.
2. I would like a solution that works with `MapView.MyLocationLayer`. This is a nice standard facility to show the user location. I want to enhance it a bit.

> The dependencies on the platforms always cause problems in the long run, and this 'native' support would also work in other platforms such as Avalonia, Uno and WinUI. Also, more platform dependent code results in more maintenance, and more bugs (because we do not test extensively on every platform).

Well, two points:
1. What I'm trying to achieve is in principle achievable with generic Mapsui facilites already (without being Mapsui.Forms-specific).
2. I'm using the Mapsui.UI.Forms.MapView, which is a nice and generic mapping solution. I think that many people use it. It supports all hardware platforms that I'm interested in.

**Comment by pauldendulk** [comment]
> I don't want to display information outside of the map (this is what the sample does that you mention, and I can already achieve that). I want information to be displayed in a label on the map itself, next to the user's location.

No, that sample shows information in the map. It is not a label but Charlennis callout, but they are similar, a callout is a kind of label but prettier because it is rendered with RichTextKit. We should use RichTextKit for all our labels. The sample also responds to a click on the feature, whatever it's shape, that is the main reason is responded.

![image](https://user-images.githubusercontent.com/963462/154051177-1505c513-f1e4-4b9c-ae74-6e5687467f41.png)

> I'm using the Mapsui.UI.Forms.MapView, which is a nice and generic mapping solution. I think that many people use it. It supports all hardware platforms that I'm interested in.

uhh..., yeah, I am aware of that 🙄. Xamarin.Forms is itself cross platform but that is irrelevant. Mapsui supports many 'implementation platforms': WPF, Xamarin.iOS, Xamarin.Android, Xamarin.Forms, Avalonia, Uno, MAUI, WinUI. This is a heavy maintenance burden. To minimize this burden we need to minimize the amount of 'implementation platform' specific code. I have been too lax with this in the past. The fact that we now have a Mapview that does not work on all platforms is a problem already. The number of 'implementation platforms' is increasing which makes it more important to reduce the platform specific code, so I intend to put more effort in it.

**Comment by janusw** [comment]
> > I don't want to display information outside of the map (this is what the sample does that you mention, and I can already achieve that). I want information to be displayed in a label on the map itself, next to the user's location.
> 
> No, that sample shows information in the map.

Oh really? That doesn't work for me on current master. Tried with Mapsui.Samples.Forms.Mac and Mapsui.Samples.Forms.Android. The "Custom Callout" sample seems to work, though.

Mapsui.Samples.Droid crashes instantly on my S21, btw:

```
Android.Views.InflateException: Binary XML file line #1 in Mapsui.Samples.Droid.Mapsui.Samples.Droid:layout/activity_main: Binary XML file line #1 in Mapsui.Samples.Droid.Mapsui.Samples.Droid:layout/content_main: Error inflating class Mapsui.UI.Android.MapControl 
```

**Comment by charlenni** [comment]
> Also, more platform dependent code results in more maintenance, and more bugs (because we do not test extensively on every platform).

But the most work is done by SkiaSharp. The only thing we have to do is the interaction of the MapControl with the underlying system. So the most useful would be to check, if we could move even more from native MapControls to the shared part.

**Comment by janusw** [comment]
@pauldendulk  @charlenni  Before this discussion drifts off too far: Could one of you possibly have a look at PR #1486, which deals with this issue here?

**Comment by janusw** [comment]
Fixed via #1486.
